src/background.ts

```diff
@@ -1,5 +1,32 @@
+let unpauseTimeout: number | null = null;
+
 chrome.action.onClicked.addListener(function (tab) {
-  void chrome.tabs.update(tab.id || 0, { url: 'https://app.graphite.dev' });
+  if (unpauseTimeout) {
+    unpause();
+  } else {
+    pause();
+    unpauseTimeout = setTimeout(() => {
+      unpause();
+    }, 5 * 60 * 1000);
+  }
 });
 
+function pause() {
+  void chrome.declarativeNetRequest.updateEnabledRulesets({
+    disableRulesetIds: ['primary_ruleset'],
+  });
+  void chrome.action.setBadgeText({ text: 'pause' });
```

Comment:
should we mention for how long?

Reply:
It's a good call - I want to do so with a custom toast, but we may need to wait a little for that :)